### PR TITLE
Migrate directory to Glimmer components

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
Navi-app was recently upgraded to octane, so we can now test migrating some of our addons

## Proposed Changes
- all glimmer components

## Breaking
- If you are overriding any directory components then it probably doesn't work anymore
- If you want to override, you must export a component from the app directory 
  > If you export just a template it will not work, you must also export a component
  > If you export just a component it will work, but probably won't be very useful since glimmer components use `this.args` a lot and it can't be modified

**Example:** To override `dir-icon` from  `navi-directory` you must have

<details>
<summary>app/app/components/dir-icon.js</summary>

```js
import DirIcon from 'navi-directory/components/dir-icon';

export default class MyDirIcon extends DirIcon {};
```
</details>

<details>
<summary>app/app/components/dir-icon.hbs</summary>

```hbs
<span>Look it's my component</span>
```
</details>

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/directory/addon/components/dir-asset-row-actions.js [hunk 1]
```diff
@@ -1,36 +1,30 @@
...
     return value ? value.constructor.modelName : null;
   }
 
   /**
    * @property {String} rowElement - selector for the row containing this
    */
-  @computed('value', 'row.rowId')
   get rowElement() {
-    const rowId = get(this, 'row.rowId');
+    const rowId = get(this, 'args.row.rowId');
```
Comment (jkusa) [inline packages/directory/addon/components/dir-asset-row-actions.js]: do we need `Ember.get`

File: packages/directory/addon/components/dir-item-name-cell.js [hunk 2]
```diff
@@ -35,26 +28,23 @@ class DirItemNameCell extends Component {
   /**
    * @property {String} itemId - the id of the model or the tempId of an unsaved model
    */
-  @readOnly('value.modelId')
-  itemId;
+  get itemId() {
+    return get(this, 'args.value.modelId');
```
Comment (jkusa) [inline packages/directory/addon/components/dir-item-name-cell.js]: is `value` required?

File: packages/directory/addon/components/dir-table-filter.js [hunk 3]
```diff
@@ -29,9 +25,9 @@ class DirTableFilter extends Component {
   /**
    * @property {String} selectedFileType
    */
-  @computed('selectedType')
+  @computed('args.selectedType')
```
Comment (jkusa) [inline packages/directory/addon/components/dir-table-filter.js]: Do we need `@computed`?

File: packages/directory/addon/components/dir-item-name-cell.js (hunk 2 above)
Comment (kevinhinterlong) [reply packages/directory/addon/components/dir-item-name-cell.js]: Hmm?

Comment (jkusa) [reply packages/directory/addon/components/dir-item-name-cell.js]: Could this be `this.args.value.modelId`?

File: packages/directory/addon/components/dir-new-button.hbs [hunk 4]
```diff
@@ -1,12 +1,12 @@
-{{!-- Copyright 2019, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+{{!-- Copyright 2020, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
 <div class="dir-new-button" ...attributes>
```
[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/directory/addon/components/dir-new-button.hbs]: BasicDropdown doesn't have a wrapper div, so we have one outside it

File: packages/directory/addon/components/dir-table.hbs [hunk 5]
```diff
@@ -1,52 +1,52 @@
-{{!-- Copyright 2019, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
+{{!-- Copyright 2020, Yahoo Holdings Inc. Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms. --}}
 {{!-- TODO: Reenable occlusion once the feature is out of beta. It currently causes issues with custom cell components --}}
 {{#light-table this.table
```
Comment (jkusa) [inline packages/directory/addon/components/dir-table.hbs]:
```suggestion
<LightTable @table={{this.table}}
```

File: packages/directory/addon/components/dir-table.hbs [hunk 6]
```diff
@@ -1,52 +1,52 @@
...
-  classNames="dir-table"
+  class="dir-table"
   responsive=true
   occlusion=false
   estimatedRowHeight=40
```
Comment (jkusa) [inline packages/directory/addon/components/dir-table.hbs]:
```suggestion
  estimatedRowHeight=40
  ...attributes
```

**Pull request merged**